kahuna/public/js/services/api/collections-api.js

```diff
@@ -0,0 +1,51 @@
+import angular from 'angular';
+import apiServices from '../api';
+
+apiServices.factory('collections', ['mediaApi', function (mediaApi) {
+    // TODO: Rx?
+    let collections;
+
+    function getCollections() {
+        if (! collections) {
+            collections = mediaApi.root.follow('collections').get().
+                then(collectionsService => collectionsService.follow('collections').get());
+        }
+        return collections;
+    }
+
+    function removeCollection(collection) {
+        return collection.perform('delete');
+    }
+
+    function addCollection(newCollectionPath) {
+        return mediaApi.root.follow('collections').post({data: newCollectionPath});
+    }
+
+    function addChildTo(node, childName) {
+        return node.perform('add-child', {body: {data: childName}}).then(childResource => {
+            const updatedChildren = node.data.children = [childResource].concat(node.data.children);
```

Comment:
Minor detail: technically, the child will be inserted in the child list using alphabetical sorting, rather than always appending? I don't think it's worth worrying about, but just wanted to point out the potential discrepancy.